### DISCUSSION TITLE
Support for MLFlow autolog feature

### DISCUSSION THREAD

**Discussion by em-pe** [Galileo-Galilei/kedro-mlflow#182] [Ideas]
Hi @Galileo-Galilei, 
I'm curious if you've considered support for [Automatic Logging](https://www.mlflow.org/docs/latest/tracking.html#automatic-logging) feature in the plugin. 

[We](https://github.com/getindata) are directing some of our customers towards `kedro` and `mlflow` marriage and this seems like a great option to showcase metadata and model registration with minimal effort - a low hanging fruit to grab. We would consider a contribution in incoming weeks / months, but only if this doesn't flash a red light on your side regarding the plugin design. 

To briefly explain the effort part -> here's the sample training code: 
```
mlflow.set_experiment("natality_tests")

with mlflow.start_run():
    mlflow.sklearn.autolog(log_models=True)
    model = sklearn.linear_model.LinearRegression()
    model.fit(natalityDF.iloc[:,2:9], natalityDF['weight'])
```

and here are the screenshots of collected metadata: https://imgur.com/a/tcVel5R

**Comment by Galileo-Galilei** [comment]
Hello @em-pe, yes this is something I plan to add and [I even have a long standing issue for this](https://github.com/Galileo-Galilei/kedro-mlflow/issues/13) ;) but I have no time for now to implement it and it does notrank high on my to do list, so feel free to take it!

However we should discuss the design in detail, because there are some conflicts risks with the `MlflowNodeHook` which logs parameters. I suggest that we continue the discussion in the issue for better tracking. I follow up tomorrow to explain how I envision the design. Feel free to make suggestions, this would be great to understand what design would be the more user friendly in your opinion.
